air-quality-backend/src/etl/in_situ/openaq_adapter.py

```diff
@@ -1,7 +1,37 @@
 from datetime import datetime
 import logging
+from src.etl.air_quality_index.calculator import (
+    get_pollutant_index_level,
+    get_overall_aqi_level,
+)
+from src.etl.air_quality_index.pollutant_type import PollutantType
 from src.etl.in_situ.sort_in_situ import sort_by_distance_and_time
 
+required_pollutant_data = {
+    "o3": PollutantType.OZONE,
+    "no2": PollutantType.NITROGEN_DIOXIDE,
+    "so2": PollutantType.SULPHUR_DIOXIDE,
+    "pm10": PollutantType.PARTICULATE_MATTER_10,
+    "pm25": PollutantType.PARTICULATE_MATTER_2_5,
+}
+
+
+def _extract_pollutants(data):
+    output = []
+    for pollutant in required_pollutant_data.keys():
+        output.append(data[pollutant]["aqi_level"])
+    return output
+
+
+def _calculate_overall_aqi_value(formatted_dataset):
+    for i in range(0, len(formatted_dataset)):
```

Comment:
We could use a for each loop instead which is more pythonic i.e.
```python
for data in formatted_dataset:
```
It stops us having to access it each time then with:
```
formatted_dataset[i]
```

Reply:
I don't think this will work as we also need to update the data formatted_dataset[i]["overall_aqi_level"] and I am pretty sure that doing this for data in formatted_dataset creates a new instance of that data so it cant be updated in the main list : )

Reply:
This isn't the case. Try this simple example in your python console:
```python
items = [{ "val": 1 }, { "val": 2 }, {"val": 3}]
for x in items:
  x["val"] = x["val"] + 1
 
print(items) # [{'val': 2}, {'val': 3}, {'val': 4}]
```

x is a reference to the object in the list you iterate through, not a new object